engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/savedGames/GameProvider.java

```diff
@@ -81,4 +87,17 @@ private GameProvider() {
         return result;
     }
 
+    private static BufferedImage getSavedGamePreviewImage(Path path) {
+        Path previewImagePath = path.resolve(TerasologyConstants.DEFAULT_GAME_PREVIEW_IMAGE_NAME);
+        if  (!previewImagePath.toFile().exists()) {
+            return null;
```

Comment:
[null is evil!](https://softwareengineering.stackexchange.com/a/12836)

Please look into the `Optional` type and replace all instances where your code returns/handles a `null` with `Optional`s - this is pretty easy to do and will save you lots of annoyances down the line.